pkgs/development/libraries/SDL2/default.nix

```diff
@@ -22,6 +22,7 @@ let
         ${if alsaSupport then "--with-alsa-prefix=${attrs.alsaLib}/lib" else ""}
         ${if (!x11Support) then "--without-x" else ""}
       '';
+  pkgConfigRequires = if (x11Support) then "x11" else "";
```

Comment:
It would be more idiomatic to use ``stdenv.lib.optionalString x11Support "x11"``.